pekf_slam_node:
  ros__parameters:
    output_frame: robot_0/odom
    base_footprint_frame: base_link
    freq: 20.0
    sensor_timeout: 1.0

    # ros topics
    odom_topic: odom
    imu_topic: mobile_base/sensors/imu_data
    pc_topic: point_cloud
    updated_pose_topic: pose_slam/updated_pose
    predicted_pose_topic: pose_slam/predicted_pose
    poses_topic: pose_slam/pose

    scan_topic: scan


    noise: 0.0
    dist_threshold: 0.25 #distance norm for updat includes angular change


    # ICP parametrs
    time_between_cloud_points: 0.2 #minimum time to wait to capture consecetive cloud points 
    transformation_epsilon: 0.001 #minimum transformation difference for termination condition
    euclidean_fitness_epsilon: 0.01 #maximum allowed Euclidean error between two consecutive steps in the ICP loop
    max_correspondence_distance: 0.05 #correspondences with higher distances will be ignored 
    max_iterations: 500 #max number of registration iterations
    ransac_iterations: 50
    ransac_threshold: 0.1

